morphic.js

```diff
@@ -8079,18 +8186,79 @@ StringMorph.prototype.mouseClickLeft = function (pos) {
     }
 };
 
+StringMorph.prototype.mouseDoubleClick = function (pos) {
+    // selects the word at pos
+    // if there is no word, we select whatever is between
+    // the previous and next words
+    var slot = this.slotAt(pos);
+
+    if (this.isEditable) {
+        this.edit();
+
+        if (slot === this.text.length) {
+            slot --;
+        }
+
+        if (isWordChar(this.text[slot])) {
+            this.selectWordAt(slot);
+        } else {
+            this.selectBetweenWordsAt(slot);
+        }
+    } else {
+        this.escalateEvent('mouseDoubleClick', pos);
+    }
+ 
+};
+
+StringMorph.prototype.selectWordAt = function (slot) {
+    var cursor = this.root().cursor;
+
+    if (slot === 0 || isWordChar(this.text[slot - 1])) {
+        cursor.gotoSlot(this.previousWordFrom(slot));
+        this.startMark = cursor.slot;
+        this.endMark = this.nextWordFrom(cursor.slot);
+    } else {
+        cursor.gotoSlot(slot);
+        this.startMark = slot;
+        this.endMark = this.nextWordFrom(slot);
+    }
+
+    this.drawNew();
+    this.changed();
+};
+
+StringMorph.prototype.selectBetweenWordsAt = function (slot) {
+    var cursor = this.root().cursor;
+
+    cursor.gotoSlot(this.nextWordFrom(this.previousWordFrom(slot)));
+    this.startMark = cursor.slot;
+    this.endMark = cursor.slot;
+
+    while (this.endMark < this.text.length
+            && !isWordChar(this.text[this.endMark])) {
+        this.endMark ++;
```

Comment:
... and another shift. I'm sorry, but it's not a big change, right... (I can do it myself, if you prefer)